sentry/sentry_test.go

```diff
@@ -0,0 +1,96 @@
+package sentry
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/fabric8-services/fabric8-wit/account"
+	"github.com/fabric8-services/fabric8-wit/login/tokencontext"
+	testtoken "github.com/fabric8-services/fabric8-wit/test/token"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/getsentry/raven-go"
+	goajwt "github.com/goadesign/goa/middleware/security/jwt"
+	"github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+)
+
+func Test_extractUserInfo(t *testing.T) {
+
+	identity := account.Identity{
+		ID:       uuid.NewV4(),
+		Username: "testuser",
+	}
+
+	tests := []struct {
+		name    string
+		ctx     context.Context
+		want    *raven.User
+		wantErr bool
+	}{
+		{
+			name:    "Given some random context",
+			ctx:     context.Background(),
+			wantErr: true,
+		},
+		{
+			name:    "fail on no token",
+			ctx:     failOnNoToken(t),
+			wantErr: true,
+		},
+		{
+			name:    "fail on parsing token",
+			ctx:     failOnParsingToken(t),
+			wantErr: true,
+		},
+		{
+			name:    "pass on parsing token",
+			ctx:     validToken(t, identity.ID.String(), identity.Username),
+			wantErr: false,
+			want: &raven.User{
+				Username: identity.Username,
+				ID:       identity.ID.String(),
+				Email:    identity.Username + "@email.com",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := extractUserInfo(tt.ctx)
```

Comment:
IMO the code will look cleaner if you just explicitly add a few Run blocks instead of going through a loop.
```
t.Run("Given some random context", func(t *testing.T) {
     got, err := extractUserInfo(context.Background())
     require.Error(t, err)
}
t.Run("Pass on parsing token", func(t *testing.T) {
     got, err := extractUserInfo(context.Background())
     require.NoError(t, err)
     require.Equal(t, &raven.User{Username: identity.Username, ID: identity.ID.String(), Email: identity.Username + "@email.com"}, got)
}

```...

Reply:
I disagree here, having it structured makes it easier to understand once you know what is the data model that is fed and expected, as opposed to the one that has no structure, each `t.Run` will be a new thing to understand.

Reply:
OK, let's agree to disagree here. But there is still no reason to use constructions like
```
if !reflect.DeepEqual(got, tt.want) {
    t.Errorf("extractUserInfo() = %v, want %v", got, tt.want)
}
```
instead of `require.Equal()`

And `require.NoError(t, err)` or `require.Error(t, err)` would look much cleaner than
```
if (err != nil) != tt.wantErr {
	t.Errorf("extractUserInfo() error = %v, wantErr %v", err, tt.wantErr)
	return
}
```

Reply:
yep done, this makes sense :-)